packages/repo-tools/src/commands/generate-catalog-info/generate-catalog-info.ts

```diff
@@ -165,8 +165,17 @@ async function fixCatalogInfoYaml(options: FixOptions) {
     relativePath('.', yamlPath),
   );
   const safeName = packageJson.name
-    .replace(/[^a-z0-9_\-\.]+/g, '-')
-    .replace(/^[^a-z0-9]|[^a-z0-9]$/g, '');
+    .replace(/^[^\w\s]|[^a-z0-9]$/g, '')
+    .replace(/[^A-Za-z0-9_\-.]+/g, '-')
+    .replace(/([A-Z])/g, (_, letter, index, original) => {
+      if (index !== 0) {
+        const previousChar = original[index - 1];
+        if (previousChar !== '-') {
+          return `-${letter.toLowerCase()}`;
+        }
+      }
+      return letter.toLowerCase();
```

Comment:
```suggestion
      return letter.toLocaleLowerCase('en-US');
```

otherwise in e.g. Turkish locales you'll get unexpected dot-less i and stuff

Reply:
I'll say though, npm package names should never be mixed case. I know this was internal, but things like https://github.com/npm/validate-npm-package-name would reject you at any point

Reply:
Yeah the mixed case ones internally are how I noticed this. There are a handful like `@internal/pluginApi` that wound up like `internal-plugin-api`